cmd/secrets_add.go

```diff
@@ -0,0 +1,158 @@
+/*
+This file is part of REANA.
+Copyright (C) 2022 CERN.
+
+REANA is free software; you can redistribute it and/or modify it
+under the terms of the MIT License; see LICENSE file for more details.
+*/
+
+package cmd
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reanahub/reana-client-go/client"
+	"reanahub/reana-client-go/client/operations"
+	"reanahub/reana-client-go/pkg/datautils"
+	"reanahub/reana-client-go/pkg/displayer"
+	"reanahub/reana-client-go/pkg/validator"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const secretsAddDesc = `
+Add secrets from literal string or from file.
+
+Examples:
+
+	$ reana-client secrets-add --env PASSWORD=password
+
+	$ reana-client secrets-add --file ~/.keytab
+
+	$ reana-client secrets-add --env USER=reanauser
+
+				   --env PASSWORD=password
+
+				   --file ~/.keytab
+`
+
+type secretsAddOptions struct {
+	token       string
+	envSecrets  []string
+	fileSecrets []string
+	overwrite   bool
+}
+
+// newSecretsAddCmd creates a command to add secrets from literal string or from file.
+func newSecretsAddCmd() *cobra.Command {
+	o := &secretsAddOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "secrets-add",
+		Short: "Add secrets from literal string or from file.",
+		Long:  secretsAddDesc,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := validator.ValidateAtLeastOne(
+				cmd.Flags(), []string{"env", "file"},
+			); err != nil {
+				return fmt.Errorf("%s\n%s", err.Error(), cmd.UsageString())
+			}
+			for _, file := range o.fileSecrets {
+				if err := validator.ValidateFile(file); err != nil {
+					return fmt.Errorf("invalid value for '--file': %s", err.Error())
+				}
+			}
```

Comment:
In order to be consistent with the Python client, this validation is done before actually reading and uploading the files. However, they could, for some reason, stop existing or have their permissions changed before uploading them. This is a concurrency edge case and, if it happens, we return the error given by `ReadFile` instead of the pretty ones from the validation